src/storage/DataAccessorBasedStore.ts

```diff
@@ -107,33 +120,35 @@ export class DataAccessorBasedStore implements ResourceStore {
     // https://solid.github.io/specification/protocol#auxiliary-resources
     await this.auxiliaryStrategy.addMetadata(metadata);
 
-    if (isContainerPath(metadata.identifier.value)) {
+    const isContainer = isContainerPath(metadata.identifier.value);
+    if (isContainer) {
       // Add containment triples of non-auxiliary resources
       for await (const child of this.accessor.getChildren(identifier)) {
         if (!this.auxiliaryStrategy.isAuxiliaryIdentifier({ path: child.identifier.value })) {
-          metadata.addQuads(child.quads());
+          if (!isMetadata) {
+            metadata.addQuads(child.quads());
+          }
           metadata.add(LDP.terms.contains, child.identifier as NamedNode, SOLID_META.terms.ResponseMetadata);
         }
       }
-
-      // Generate a container representation from the metadata
-      // All triples should be in the same graph for the data representation
-      const data = metadata.quads().map((triple): Quad => {
-        if (triple.graph.termType === 'DefaultGraph') {
-          return triple;
-        }
-        return DataFactory.quad(triple.subject, triple.predicate, triple.object);
-      });
+      const data = metadata.quads();
+      if (isMetadata) {
+        metadata = new RepresentationMetadata(this.metadataStrategy.getAuxiliaryIdentifier(identifier));
+      }
 
       metadata.addQuad(DC.terms.namespace, PREFERRED_PREFIX_TERM, 'dc', SOLID_META.terms.ResponseMetadata);
       metadata.addQuad(LDP.terms.namespace, PREFERRED_PREFIX_TERM, 'ldp', SOLID_META.terms.ResponseMetadata);
       metadata.addQuad(POSIX.terms.namespace, PREFERRED_PREFIX_TERM, 'posix', SOLID_META.terms.ResponseMetadata);
       metadata.addQuad(XSD.terms.namespace, PREFERRED_PREFIX_TERM, 'xsd', SOLID_META.terms.ResponseMetadata);
       representation = new BasicRepresentation(data, metadata, INTERNAL_QUADS);
-    } else {
-      // Retrieve a document representation from the accessor
+    } else if (!isMetadata) {
       representation = new BasicRepresentation(await this.accessor.getData(identifier), metadata);
+    } else {
+      representation = new BasicRepresentation(
+        metadata.quads(), this.metadataStrategy.getAuxiliaryIdentifier(identifier), INTERNAL_QUADS,
+      );
```

Comment:
Would still do the split I suggested in https://github.com/CommunitySolidServer/CommunitySolidServer/pull/1188#discussion_r857432418 so you also have the prefix triples when GETting the metadata of a document. These cause the resulting turtle to be prettier.